File: Printpress.UI/Printpress_Angular/src/app/features/reports/models/cash-flow-report.dto.ts
export interface CashFlowReportDto {
  dateFrom: string | null;
  dateTo: string | null;
  cashAccountId: string | null;
  totalIn: number;
  totalOut: number;
  net: number;
  byDay: CashFlowBucketDto[];
  byMonth: CashFlowBucketDto[];
}

export interface CashFlowBucketDto {
  key: string;
  label: string;
  periodStart: string | null;
  periodEnd: string | null;
  totalIn: number;
  totalOut: number;
  net: number;
  transactionCount: number;
}

File: Printpress.Application/Reports/DTO/CashFlowReportDto.cs
namespace Printpress.Application;

public class CashFlowReportDto
{
    public DateTime? DateFrom { get; set; }
    public DateTime? DateTo { get; set; }
    public Guid? CashAccountId { get; set; }
    public decimal TotalIn { get; set; }
    public decimal TotalOut { get; set; }
    public decimal Net { get; set; }
    public List<CashFlowBucketDto> ByDay { get; set; } = [];
    public List<CashFlowBucketDto> ByMonth { get; set; } = [];
}

public class CashFlowBucketDto
{
    public string Key { get; set; }
    public string Label { get; set; }
    public DateTime? PeriodStart { get; set; }
    public DateTime? PeriodEnd { get; set; }
    public decimal TotalIn { get; set; }
    public decimal TotalOut { get; set; }
    public decimal Net { get; set; }
    public int TransactionCount { get; set; }
}
